### ISSUE TITLE
typescript.autoImportSuggestions should be off by default

### ISSUE THREAD

**Issue by csicky** [Microsoft/vscode#57822]
Please consider having

`typescript.autoImportSuggestions.enabled` 

off by default. 

Reported in other places already, this is so annoying and wastes so much time to find.
If anybody wants it, should enable it. For the rest of us it's a very annoying and time wasting surprise. 

https://github.com/Microsoft/vscode/issues/40001
https://github.com/Microsoft/vscode/issues/38911

If it's auto imported, why throws error? 
`"TypeError: timeout.close is not a function"`

A JavaScript developer writes `clearTimeout` and doesn't think twice about it. `timeout.close` is not an expected error. If it's imported by default it should also work just like the original it shadows. But still, auto-import is dangerous. In this case it overwrites existing language feature. **VS Code is not in position to change JavaScript.** In a TypeScript project, maybe, but the user should know that an auto-import happened. Also the setting name starts with `typescript`, so default true in a TypeScript project.

Steps to Reproduce:
Use a `clearTimeout` in a Vue.js project.

Does this issue occur when all extensions are disabled?: Yes

- VSCode Version: 1.26.1
- OS Version: Windows 10

**Comment by mjbvz** [comment]
The problem is that are adding imports for `clearTimeout` and other globals. Your proposed solution of disabling auto imports entirely is not reasonable as it's been over a year since we added this and people expect auto imports to just be there. 

What is the exact code you wrote that triggered this? Were you in a `.vue` file or a `.js`/`.ts` file? If you run `go to definition` on `clearTimeout`, what file does it open?

**Comment by csicky** [comment]
To be honest I also happen to have auto import being useful sometimes. When writing `...mapState` it imports it from `vuex` (most of the time correctly, but also happens to place the import inside the template).  Auto import is a good feature, just needs to be fine tuned.

Problem is when it imports things which overwrites/shadows language features. Basically anything that exists on the `window` object it's expected to just work without me writing `window.clearTimeout`, but simply `clearTimeout`. Now there are problematic auto imports which overwrite language features which is very bad. 

Would be also useful if some kind of toast message would show up saying, hey we just imported this for you, then to disappear after  some seconds. People would know that an auto import happened and would waste much less time when an auto import misbehaved. 

Specifically my code which triggered an auto import of some `timeout` object and wasted some half an hour for me was the following, in a `.vue` file:

```
      clearTimeout(this.timeoutId)
      this.timeoutId = setTimeout(() => {
        //do something here only once
      }, 300)
```

**Comment by mjbvz** [comment]
Thanks for the details.

Our built-in js/ts support is not actually used for `.vue` files so I suggest that you file this issue against the Vue extension instead. I had trouble reproducing this in a js/ts file and I suspect that the vue extension may be using an older version of TypeScript that had this bug for auto imports.

Please let me know if you also see this error in js/ts files when using TypeScript 3.0

**Comment by octref** [comment]
This is an issue that exists for both Vue and TS and needs to be addressed in TS side, we pick up the latest TS Language Service.

> Problem is when it imports things which overwrites/shadows language features. Basically anything that exists on the window object it's expected to just work without me writing window.clearTimeout, but simply clearTimeout. Now there are problematic auto imports which overwrite language features which is very bad.

![image](https://user-images.githubusercontent.com/4033249/45110084-2ab8d600-b0f6-11e8-993c-0250bdf23e04.png)

@mjbvz 

The `timers` package comes from `@types/node` (for https://nodejs.org/api/timers.html). However, they are globally available:

> The timer module exposes a global API for scheduling functions to be called at some future period of time. Because the timer functions are globals, there is no need to call require('timers') to use the API.

**Comment by octref** [comment]
So I suggest either reopening this one or https://github.com/Microsoft/vscode/issues/38911.

**Comment by mjbvz** [comment]
What version of TS are you using in Vetur / how are you getting ts intellisense in vue files?

**Comment by octref** [comment]
I'm in an empty folder with only one file `foo.ts`:

![image](https://user-images.githubusercontent.com/4033249/45110983-a1ef6980-b0f8-11e8-9dda-c1cadcbd2533.png)

**Comment by csicky** [comment]
The problem is then that too many things are globally available. That is another pain in using VS Code. Most of the times I have to press `Esc` to avoid all kinds of unneeded and unheard of autocompletes or if I forget to press `Esc` I have to `Ctrl + Z` to undo it. I always wondered who and why did import into global namespace so many useless things. There should be only what exists normally on `window` and what is available in existing component I work on. Of course if anybody wants all kinds of things available that is up to them. 
For example I type `sett` in a Vue component. I get suggestions I don't need, I don't know where are coming from and do more problem instead of help. What is `ClusterSettings`, `setupMaster`, etc.
Then I go to Chrome console and type the same `sett` and I don't see those exotic things.

Combine useless suggestions with auto import and we get current situation where people must spend time investigating problems which are not created by them.

**tldr:
What I can see in any browser console + what is available from my code and my settings should be available in suggestions list, nothing else.**
